pcsx2/IopBios.cpp

```diff
@@ -1387,3 +1415,57 @@ namespace R3000A
 	}
 
 } // end namespace R3000A
+
+bool SaveStateBase::handleFreeze()
+{
+	if (!EmuConfig.HostFs) //if hostfs isn't enabled, skip loading/saving file handles
+		return IsOkay();
+
+	if (IsLoading())
+		R3000A::ioman::reset();
+
+	if (!FreezeTag("hostHandles"))
+		return false;
+
+	const int firstfd = R3000A::ioman::firstfd;
+	int handleCount = R3000A::handles.size();
+	Freeze(handleCount);
+
+	for (int i = 0; i < handleCount; i++)
+	{
+		if (IsLoading())
+		{
+			//load the parameters for opening the file
+			long pos;
+			Freeze(pos);
+
+			R3000A::fileHandle handle;
+			Freeze(handle.flags);
+			FreezeString(handle.full_path);
+			Freeze(handle.mode);
+			R3000A::handles.push_back(handle);
+
+			//reopen the file
+			IOManFile* file = NULL;
+			R3000A::HostFile::open(&file, handle.full_path, handle.flags, handle.mode);
```

Comment:
Can you check the return of this open call or check if `file` is `nullptr`? It's probably best to remove the handle if this is the case. 
You can run into issues if you load a state after renaming / removing a file that was previously saved.

Also I'd add a warning in the console with the file path via `Console.Warning(...)` so people know why their state isn't working as intended.